File: C#/TddApi/NavigationApp/s18.inject_provider/Navigator.cs
﻿using System;

namespace NavigationApp.s18.inject_provider
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;
        Location destination;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, destination);
        }

        public void setDestination(Location dest)
        {
            destination = dest;
        }
    }
}

File: Python/s16_pass_unit_test/navigator.py
class Navigator:
    def __init__(self, starting_point, dist_provider):
        self.dist_provider = dist_provider
        self.starting_point = starting_point
        self.destination = None

    def getDistanceFromDestination(self):
        return self.dist_provider.get_distance(self.starting_point, self.destination)

    def setDestination(self, dest):
        self.destination = dest
